### PR TITLE
Optional consistency check between owner and student_id upon collect

### PR DESCRIPTION
**Pull request by nthiery**
Mitigates #1097, #1098.

### CODE REVIEW

**Comment by nthiery** [comment]
Hmm, this may require a  nowindows decorator. Let's see what the tests say.

File: nbgrader/exchange/collect.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import os
 import glob
+import pwd
```
Comment (jhamrick) [inline nbgrader/exchange/collect.py]: Ah yeah, I think you will need to wrap this in a try/except so that the file can at least be imported on windows.

Comment (jhamrick) [reply nbgrader/exchange/collect.py]: I think you still need to remove this line as it's still trying to import `pwd` on windows :(

Comment (nthiery) [reply nbgrader/exchange/collect.py]: Hem, hem, hem :-) Sounds like I was not quite awake. Thank you for the cleanup!

**Comment by nthiery** [comment]
Thank you Jessica for the review!

File: nbgrader/exchange/collect.py (hunk 1 above)
Comment (jhamrick) [reply nbgrader/exchange/collect.py]: It happens 😄No problem!

**Pull request merged**